### PR TITLE
Editlab + CreateLab Progress

### PR DESCRIPTION
**Pull request by YazanSalem**
Wrote views with edit lab and create lab using lab management classes

Completed CreateLab html, editlab html has a placeholder page as of right now

urls.py also updated to include editLab and createLab

Please let me know if you see anything wrong/typos

### CODE REVIEW

**Review by WajahatAliShah1 — COMMENTED** [review]
Looks good!, will update mine to see if it works, then we can merge.

File: TAScheduler/views.py [hunk 1]
```diff
@@ -135,6 +139,32 @@ def get(self, request):
...
+    def get(self, request):
+        # If the user does not have a valid name, I.E. if they try to manually enter /home in the search bar,
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            LabManagement.createLab(request, request.POST['labID'], request.POST['labName'],
+            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
+            request.POST['labInstructor'], request.POST['labTA'])
+        else:
+            return redirect("/../home/")
```
Comment (BMHebert) [inline TAScheduler/views.py]:
```suggestion
        if userAllowed(request, ["SUPERVISOR"]):
            return render(request, "createlab.html")
        else:
            return redirect("/../home/")
            
def post(self, request):
     LabManagement.createLab(request, request.POST['labID'], request.POST['labName'],
            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
            request.POST['labInstructor'], request.POST['labTA'])
            return render(request, "createlab.html")
```
Form data needs to be processed in post whereas get handles the redirects and rendering of the page.

File: TAScheduler/views.py [hunk 2]
```diff
@@ -135,6 +139,32 @@ def get(self, request):
...
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            LabManagement.editLab(request, request.POST['labID'], request.POST['labName'],
+            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
+            request.POST['labInstructor'], request.POST['labTA'])
+            
+            return render(request, "editlab.html")
+        else:
+            return redirect("/../home/")
```
Comment (BMHebert) [inline TAScheduler/views.py]:
```suggestion
        if userAllowed(request, ["SUPERVISOR"]):
            return render(request, "editlab.html")
        else:
            return redirect("/../home/")
            
def post(self, request):
     LabManagement.editLab(request, request.POST['labID'], request.POST['labName'],
            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
            request.POST['labInstructor'], request.POST['labTA'])
            return render(request, "editlab.html")
```
Same as above

File: TAScheduler/Managment/CourseManagement.py [hunk 3]
```diff
@@ -3,8 +3,12 @@
 # deleteCourse(courseName)
 # populateSearchClass(searchPromp)  should change class to course
 # displayAllCourse()
-
-class CourseManagement:
+from types import coroutine
+from django.db import models
+from TAScheduler.models import Course
+from TAScheduler.models import UserProfile
```
Comment (BMHebert) [inline TAScheduler/Managment/CourseManagement.py:9]:
```suggestion
from TAScheduler.models import Course, UserProfile
```
Only needs one line

**Review by BMHebert — CHANGES_REQUESTED** [review]
The views.py file had a few issues with CreateLab and EditLab not having post methods. I've suggested fixes for each of these issues. Other than that, there were a few small suggestions I had for slight ease of readability, but as a whole it looks good.

**Review by BMHebert — APPROVED** [review]

**Pull request merged**